app-config-electron/src/index.ts

```diff
@@ -0,0 +1,28 @@
+import type { WebPreferences } from 'electron';
+import type { ExportedConfig } from '@app-config/main';
+import * as path from 'path';
+
+export function addAppConfigPreload(config: ExportedConfig, baseWebPreferences?: WebPreferences) {
```

Comment:
I'm guessing there aren't great ways to unit test this?

[This](https://www.electronjs.org/spectron) looks interesting, I can help if you'd like

Reply:
I gave it a try but didn't get far. Maybe you could take a look? I can't get it to start, I get: `Failed to create session. unknown error: Chrome failed to start: exited abnormally. (unknown error: DevToolsActivePort file doesn't exist)` 

Seems like it might have something to do with spectron not supporting `contextIsolation: true` which we need for the preload script. See https://github.com/electron-userland/spectron/issues/834 and https://github.com/electron-userland/spectron/issues/693.

Reply:
Nevermind I figured it out. Apparently Spectron doesn't support newer versions of electron very well and playwright works much better. Not sure if they'll work in CI right away though, we might need to do some xvfb stuff or something

Reply:
FYI this isn't triggering CI builds because it's from your fork. I just updated upstream to run on pull_request, maybe try pulling master?

Reply:
Ah ok, seems like it's working now. You need to approve it though

Reply:
:+1: done

Reply:
Thanks! Looks like it tried to do a docker-image/publish? Probably don't want to do that for PRs

Reply:
Already fixed upstream, noticed that too, oops